openmc/mgxs/mdgxs.py

```diff
@@ -785,40 +882,53 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
         else:
             df = df.drop('score', axis=1)
 
-        # Override energy groups bounds with indices
-        all_groups = np.arange(self.num_groups, 0, -1, dtype=np.int)
-        all_groups = np.repeat(all_groups, len(query_nuclides))
-        if 'energy low [eV]' in df and 'energyout low [eV]' in df:
-            df.rename(columns={'energy low [eV]': 'group in'},
-                      inplace=True)
-            in_groups = np.tile(all_groups, int(self.num_subdomains *
-                                                self.num_delayed_groups))
-            in_groups = np.repeat(in_groups, int(df.shape[0] / in_groups.size))
-            df['group in'] = in_groups
-            del df['energy high [eV]']
+        # Override polar and azimuthal bounds with indices
+        if self.num_polar or self.num_azimuthal:
+            # First for polar
+            del df['polar high']
+            df.rename(columns={'polar low': 'polar bin'}, inplace=True)
+            df_bins = df['polar bin']
+            if self.num_polar:
+                pol_bins = np.linspace(0., np.pi,
+                                       num=self.num_polar + 1,
+                                       endpoint=True)
+            else:
+                pol_bins = np.linspace(0., np.pi, num=2, endpoint=True)
+            df['polar bin'] = np.searchsorted(pol_bins, df_bins) + 1
 
-            df.rename(columns={'energyout low [eV]': 'group out'},
+            # Second for azimuthal
+            del df['azimuthal high']
+            df.rename(columns={'azimuthal low': 'azimuthal bin'},
                       inplace=True)
-            out_groups = np.repeat(all_groups, self.xs_tally.num_scores)
-            out_groups = np.tile(out_groups, int(df.shape[0] / out_groups.size))
-            df['group out'] = out_groups
-            del df['energyout high [eV]']
-            columns = ['group in', 'group out']
+            df_bins = df['azimuthal bin']
+            if self.num_azimuthal:
+                azi_bins = np.linspace(-np.pi, np.pi,
+                                       num=self.num_azimuthal + 1,
+                                       endpoint=True)
+            else:
+                azi_bins = np.linspace(-np.pi, np.pi, num=2,
```

Comment:
Coming back to this, I presume you would argue that `np.linspace` and `np.searchsorted` is actually simpler than the previous convoluted logic for energy group indices with `np.tile` and `np.repeat`?